src/main/java/com/github/resalner/javapractice/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.github.resalner.javapractice.service;
+
+import com.github.resalner.javapractice.model.Category;
+import com.github.resalner.javapractice.repository.CategoryRepository;
+import com.github.resalner.javapractice.request.CategoryRequest;
+import com.github.resalner.javapractice.exception.EntityNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+@Service
+public class CategoryServiceImpl {
+
+  @Autowired
+  private final CategoryRepository categoryRepository;
+  
+
+  public CategoryService(CategoryRepository categoryRepository){
+    this.categoryRepository = categoryRepository;
+  }
+
+  public List<Category> getCategories(){
+    return categoryRepository.findAll();
+  }
+
+  public void addCategory(CategoryRequest categoryRequest){
+    Category category = new Category();
+    category.setName(categoryRequest.name());
+    categoryRepository.save(category);
+    
+  }
+  public Category getCategory(long id){
+    return categoryRepository.findById(id).orElseThrow(()->EntityNotFoundException("не найдена категория с id = "+id));
+  }
```

Comment:
Не забывай прожимать ctrl + alt + L , он автоматом расставит нужные пробелы
```suggestion
    return categoryRepository.findById(id)
    .orElseThrow(() -> EntityNotFoundException("не найдена категория с id = " + id));
  }
```